features/facebookImport/src/components/advertisingValueMiniStory/advertisingValueMiniStory.jsx

```diff
@@ -1,9 +1,31 @@
-import React from "react";
+import React, { useRef } from "react";
 import i18n from "../../i18n";
 
 import "./advertisingValueMiniStory.css";
 
+const calculateFontSize = (text, maxWidth) => {
+    // TODO: Extract text size affecting styles from target element
+    const minFontSize = 14;
+    const maxFontSize = 34;
+    const canvas = document.createElement("canvas");
+    const context = canvas.getContext("2d");
+    for (let fontSize = maxFontSize; fontSize > minFontSize; fontSize--) {
+        context.font = `${fontSize}px Jost`;
+        if (context.measureText(text).width <= maxWidth) return fontSize;
+    }
+    return minFontSize;
+};
+
 const AdvertisingValueMiniStory = ({ randomAdInterests, numberInterests }) => {
+    // following takes the three interests, splits them into single words and picks the longest word
+    let splitInterests = [].concat.apply(
+        [],
+        randomAdInterests.map((int) => int.split(" "))
+    );
+    let longestInterestWord = splitInterests.sort(function (a, b) {
```

Comment:
If you're interested in the min or max of a list, you don't need to sort; sorting is O(n²) while max/min is simply O(n). Not a big deal generally, but imagine we had like 1000 words here.

Reply:
Will push soonish your suggestion... Please have a look then. I'm not sure if it's weird. But it works as well.

Reply:
Feel freee to comment on my latest commit